### PR TITLE
Dipoles to evoked

### PR DESCRIPTION
**Pull request by agramfort**
taking over #2846

### CODE REVIEW

File: doc/whats_new.rst [hunk 1]
```diff
@@ -34,6 +34,9 @@ Changelog
 
     - Add command ``mne show_info`` to quickly show the measurement info from a .fif file from the terminal by `Alex Gramfort`_.
 
+    - Add creating forward operator for dipole object :func:`mne.forward.make_forward_dipole` by `Chris Bailey`_
+
+
```
Comment (Eric89GXL) [inline doc/whats_new.rst]: two spaces????

File: mne/forward/_make_forward.py [hunk 2]
```diff
@@ -605,6 +612,130 @@ def make_forward_solution(info, trans, src, bem, fname=None, meg=True,
     return fwd
 
 
+def make_forward_dipole(dipole, bem, info, trans=None, n_jobs=1, verbose=None):
+    """Convert dipole object to source estimate and calculate forward operator
+
+    The instance of Dipole is converted to a discrete volume source space,
```
Comment (Eric89GXL) [inline mne/forward/_make_forward.py]: just "discrete source space", not "discrete volume source space"

File: mne/forward/_make_forward.py [hunk 3]
```diff
@@ -605,6 +612,130 @@ def make_forward_solution(info, trans, src, bem, fname=None, meg=True,
...
+        inuse = fwd['src'][0]['inuse'].astype(np.bool)
+        head = ('The following dipoles are outside the inner skull boundary')
+        msg = len(head) * '#' + '\n' + head + '\n'
+        for (t, pos) in zip(times[np.logical_not(inuse)],
+                            pos[np.logical_not(inuse)]):
+            msg += '    t={:.0f} ms, pos=({:.0f}, {:.0f}, {:.0f}) mm\n'.\
+                format(t * 1000., pos[0] * 1000.,
+                       pos[1] * 1000., pos[2] * 1000.)
+        msg += len(head) * '#'
+        print(msg)
```
Comment (Eric89GXL) [inline mne/forward/_make_forward.py]: logger.error

File: mne/source_space.py [hunk 4]
```diff
@@ -1549,19 +1549,28 @@ def _make_voxel_ras_trans(move, ras, voxel_size):
...
 
     Returns
     -------
     src : dict
         The source space.
     """
+    # Check that coordinate frame is valid
+    if coord_frame not in _str_to_frame:  # will fail if coord_frame not string
+        raise KeyError('coord_frame must be one of %s, not "%s"'
+                       % (list(_str_to_frame.keys()), coord_frame))
```
Comment (Eric89GXL) [inline mne/source_space.py]: we should refactor `_check_frame` from `mne/channels/montage.py`, and move it to `mne/transforms.py` in the process

**Comment by Eric89GXL** [comment]
Otherwise LGTM. Let me know when I should run the example

**Comment by agramfort** [comment]
comments addressed

**Comment by Eric89GXL** [comment]
Leaving the bit of refactoring to a follow-up PR?

**Comment by agramfort** [comment]
yes

**Comment by Eric89GXL** [comment]
You on it or should I do it after this is merged?

File: examples/inverse/plot_dipole_fit.py [hunk 5]
```diff
@@ -39,5 +50,33 @@
...
+
+# compare this to the predicted field
+pred_evoked.plot_topomap(time_format='Predicted field', axes=axes[1],
+                         **plot_params)
+
+# Subtract predicted from measured data (apply equal weights)
+diff = combine_evoked([evoked, pred_evoked], [1, -1])
+plot_params['colorbar'] = True
+diff.plot_topomap(time_format='Difference', axes=axes[2], **plot_params)
+plt.suptitle('Comparison of measured and predicted fields '
```
Comment (Eric89GXL) [inline examples/inverse/plot_dipole_fit.py]: use the `title` kwarg of `plot_topomap`

Comment (agramfort) [inline examples/inverse/plot_dipole_fit.py]: using title would not render as nice for such a custom plot.

**Comment by Eric89GXL** [comment]
Thanks @agramfort

**Pull request merged**